webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../common/I18n';
+
+const LoginPageFooter = ({ caption }) => {
+  const [showLess, setShowLess] = React.useState(true);
+  if (!caption) return null;
+  let loginCaption = caption;
+
+  if (caption.length > 500) {
+    loginCaption = (
+      <React.Fragment>
+        {showLess ? `${caption.substring(0, 500)}...` : caption}
+        <Button
+          className="btn-expand"
+          variant="link"
+          onClick={() => setShowLess(!showLess)}
```

Comment:
state setters functions are async (and are not activate imminently), therefore we can't trust the current state value.
This approach is safer: 
```js
setShowLess(prevState => !prevState)
```